rs/index/src/segment/mutable_segment.rs

```diff
@@ -4,13 +4,17 @@ use std::time::Instant;
 use anyhow::{Ok, Result};
 use config::collection::CollectionConfig;
 use log::debug;
+use std::collections::{HashSet, HashMap};
+use parking_lot::{RwLock};
 
 use crate::multi_spann::builder::MultiSpannBuilder;
 use crate::multi_spann::writer::MultiSpannWriter;
 
 pub struct MutableSegment {
     multi_spann_builder: MultiSpannBuilder,
 
+    use_invalidated_id_map: bool,
```

Comment:
does this need to be Atomic or something?

Reply:
I think let's called it `flushing_started` the main purpose of this is to signal that flushing has started